invertedai/large/initialize.py

```diff
@@ -0,0 +1,265 @@
+import time
+from pydantic import BaseModel, validate_call
+from typing import List, Optional, Tuple
+from itertools import product
+from tqdm.contrib import tenumerate
+import numpy as np
+
+import invertedai as iai
+from invertedai.large.common import Region
+from invertedai.api.initialize import InitializeResponse
+from invertedai.common import TrafficLightStatesDict, Point
+from invertedai.utils import get_default_agent_attributes
+from invertedai.error import InvertedAIError
+
+SLACK = 2
+AGENT_SCOPE_FOV_BUFFER = 20
+
+@validate_call
+def define_regions_grid(
+    map_center: Optional[Tuple[float,float]] = (0.0,0.0),
+    width: Optional[float] = 100.0, 
+    height: Optional[float] = 100.0, 
+    stride: Optional[float] = 50.0
+) -> List[Region]:
+    def check_valid_center(center):
+        return (map_center[0] - width) < center[0] < (map_center[0] + width) and \
+            (map_center[1] - height) < center[1] < (map_center[1] + height)
+
+    def get_neighbors(center):
+        return [
+            (center[0] + (i * stride), center[1] + (j * stride))
+            for i, j in list(product(*[(-1, 1),]* 2))
+        ]
+
+    queue, centers = [map_center], []
+
+    while queue:
```

Comment:
BFS is way of an overkill here. You can compute the grid of a rectangular area in closed form.

Reply:
For the sake of time, I can update this later since this won't impact the user interface and we know this code works so we can quietly make that change later while users are testing the features?

Reply:
Yes, it can be done later.